code/game/objects/structures.dm

```diff
@@ -93,12 +94,17 @@
 	if(!can_touch(user) || !climbable)
 		return FALSE
 
-	user.forceMove(get_turf(src))
-	if(get_turf(user) == get_turf(src))
+	return TRUE
+
+/obj/structure/proc/start_climb(mob/living/user)
+	climbers += user
```

Comment:
register a signal here to listen if the user gets deleted please, and remove them from the list if they do